README.md

```diff
@@ -154,9 +140,52 @@ lspconfig.util.default_config = vim.tbl_extend(
 )
 ```
 
+## Individual server settings and initialization options
+
+See [CONFIG.md](CONFIG.md) for documentation and configuration of individual language servers.
+This document contains installation instructions for each language server, and is 
+auto-generated from the documentation in the lua source. Do not submit PRs 
+modifying CONFIG.md directly; CONFIG.md will be overwrriten by docgen
+
+**You do not need to copy settings or init_options from this configuration into your config**
+
+## Keybindings and completion
+
+The following maps most of the standard functions to keybindings, and maps omnicomplete to use
+the lsp.omnifunc. See `:help lsp for more details`
+```vim
+lua << EOF
+local nvim_lsp = require('lspconfig')
+local on_attach = function(_, bufnr)
+  vim.api.nvim_buf_set_option(bufnr, 'omnifunc', 'v:lua.vim.lsp.omnifunc')
+
+  local opts = { noremap=true, silent=true }
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', 'gD', '<Cmd>lua vim.lsp.buf.declaration()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', 'gd', '<Cmd>lua vim.lsp.buf.definition()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', 'K', '<Cmd>lua vim.lsp.buf.hover()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', 'gi', '<cmd>lua vim.lsp.buf.implementation()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<C-k>', '<cmd>lua vim.lsp.buf.signature_help()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<leader>wa', '<cmd>lua vim.lsp.buf.add_workspace_folder()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<leader>wr', '<cmd>lua vim.lsp.buf.remove_workspace_folder()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<leader>wl', '<cmd>lua print(vim.inspect(vim.lsp.buf.list_workspace_folders()))<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<leader>D', '<cmd>lua vim.lsp.buf.type_definition()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<leader>rn', '<cmd>lua vim.lsp.buf.rename()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', 'gr', '<cmd>lua vim.lsp.buf.references()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<leader>e', '<cmd>lua vim.lsp.diagnostic.show_line_diagnostics()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '[d', '<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', ']d', '<cmd>lua vim.lsp.diagnostic.goto_next()<CR>', opts)
+  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<leader>q', '<cmd>lua vim.lsp.diagnostic.set_loclist()<CR>', opts)
+end
```

Comment:
Same comment here ^

https://github.com/neovim/nvim-lspconfig/pull/517#discussion_r553183906

Reply:
Yeah, I pushed before I saw yours. Do you think this is a good idea to put in the readme? The biggest issue people are saying is that it's hard to configure, so I imagine having some sort of "sane defaults" mentioned somewhere would be good, instead of being buried in issues.

Reply:
I don't see how sane defaults and making the examples terser by reducing some redundancy are related? I think we should provide nice examples since this is going to be copied _a lot_. But then again, "nice" is in the eye of the beholder.

Reply:
I'm fine with skipping this though to keep the example free of too many abstractions/indirections.

Reply:
Oh no no, that's not what I meant haha. I like your change. I meant what *I* included, independently of using the helper function.

Reply:
Sorry for the miscommunication.

Reply:
> Oh no no, that's not what I meant haha. I like your change. I meant what _I_ included, independently of using the helper function.

Oooh! I thought you were referring to my helper function. Not the examples themselves. Anyway, I think it's a good idea to include some sample configuration including key mappings for the user. I assume coc.nvim and vim-lsp added it to their readme for similar reasons.

> Sorry for the miscommunication.

No worries! 🙂